platform/pulpcore/app/serializers/content.py

```diff
@@ -23,9 +24,13 @@ class Meta:
 
 
 class ArtifactSerializer(base.ModelSerializer):
+    _href = serializers.HyperlinkedIdentityField(
+        view_name='artifacts-detail',
+    )
+
     file = serializers.FileField(
         help_text=_("The stored file."),
-        read_only=True
+        required=True
```

Comment:
What does this actually mean for a ``FileField``? That bits must be present? Note that for on-demand content, we will need an ``Artifact`` record so that we can publish it and also have ``Catalog`` entries, but the file may not yet be present.